pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,105 @@
+package node
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var notReadyTolerationSecondsPath = []string{"apiServerArguments", "default-not-ready-toleration-seconds"}
+var unreachableTolerationSecondsPath = []string{"apiServerArguments", "default-unreachable-toleration-seconds"}
+
+// ObserveNotReadyTolerationSeconds observes the value that should be set for default-not-ready-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveNotReadyTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
```

Comment:
I think you will need similar logic for `kube-controller-manager` and `kube-scheduler` operators. I think we could write a generic observer that could be placed in `library-go`.

Reply:
We're in library-go. (openshift/library-go#1364)